protostar/migrator/migrator_contract_identifier_resolver.py

```diff
@@ -13,20 +27,19 @@ def __init__(
         self._project_compiler = project_compiler
         self._migrator_datetime_state = migrator_datetime_state
 
-    def resolve(self, contract_identifier: str) -> Path:
+    def resolve(self, contract_identifier: str, build_output_dir: Path) -> Path:
```

Comment:
[Design] `build_output_dir_path` can be injected directly to this class through constructor.

Reply:
done